core/client-internal/src/test/java/alluxio/hadoop/HdfsFileInputStreamTest.java

```diff
@@ -0,0 +1,173 @@
+import alluxio.exception.ExceptionMessage;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.FileSystem.Statistics;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.io.IOException;
+import java.util.Random;
+
+/**
+ * Unit tests for {@link HdfsFileInputStream}.
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({HdfsFileInputStream.class, AlluxioURI.class,
+    Path.class, Configuration.class, Statistics.class, FileSystem.class})
+public class HdfsFileInputStreamTest {
+  @Rule
+  public final ExpectedException mThrown = ExpectedException.none();
+
+  private static final int BUFSIZE = 1024;
+  private static Random sRandom = new Random();
+
+  private HdfsFileInputStream mHdfsFileInputStream;
+  private AlluxioURI mAlluxioURI;
+  private Path mHdfsPath;
+  private org.apache.hadoop.conf.Configuration mHdfsConf;
+  private org.apache.hadoop.fs.FileSystem.Statistics mStats;
+  private URIStatus mFileInfo;
+  private FileInStream mAlluxioFileInputStream = null;
+
+  /**
+   * Sets up the context before a test runs.
+   *
+   * @throws Exception when the initialize fail
+   */
+  @Before
+  public void before() throws Exception {
+    mAlluxioURI = PowerMockito.mock(AlluxioURI.class);
+    mHdfsPath = PowerMockito.mock(Path.class);
+    mHdfsConf = PowerMockito.mock(org.apache.hadoop.conf.Configuration.class);
+    mStats = PowerMockito.mock(org.apache.hadoop.fs.FileSystem.Statistics.class);
+    mFileInfo = PowerMockito.mock(URIStatus.class);
+    mAlluxioFileInputStream = PowerMockito.mock(FileInStream.class);
+
+    PowerMockito.mockStatic(FileSystem.Factory.class);
+    FileSystem fs = PowerMockito.mock(FileSystem.class);
+    when(FileSystem.Factory.get()).thenReturn(fs);
+
+    when(fs.getStatus(mAlluxioURI)).thenReturn(mFileInfo);
+    when(fs.openFile(any(AlluxioURI.class), any(OpenFileOptions.class)))
+      .thenReturn(mAlluxioFileInputStream);
+
+    mHdfsFileInputStream = new HdfsFileInputStream(
+        mAlluxioURI, mHdfsPath, mHdfsConf, BUFSIZE, mStats);
+  }
+
+  /**
+   * Resets the test context.
+   *
+   * @throws IOException when the initialize fail
+   */
+  @After
+  public void after() throws IOException {
+    mHdfsFileInputStream.close();
+  }
+
+  /**
+   * Test the {@link available()} method.
+   *
+   * @throws IOException Should throw IOException
+   */
+  @Test
+  public void availableTest() throws IOException {
+    mThrown.expect(IOException.class);
+    mThrown.expectMessage(ExceptionMessage.NOT_SUPPORTED.getMessage());
+    mHdfsFileInputStream.available();
+    verify(mHdfsFileInputStream).available();
```

Comment:
this line is never reached because of the thrown exceptoin - similar for other tests